gen/target.cpp

```
@@ -46,9 +47,11 @@ llvm::Type *getRealType(const llvm::Triple &triple) {
 
   auto &ctx = getGlobalContext();
 
-  // If user specified double precision, use it unconditionally.
-  if (realTypeEncoding == RealTypeEncoding::Double) {
+  // If user specified double or quad precision, use it unconditionally.
+  if (realPrecision == RealPrecision::Double) {
     return LLType::getDoubleTy(ctx);
+  } else if (realPrecision == RealPrecision::Quad) {
+    return LLType::getFP128Ty(ctx);
```

Have you tested if this will blow up x86 builds?

---

Android x86_64 is already using it, so I guess that's fine. - This is an advanced option (hence hidden), for pros only - druntime and Phobos need to match too etc. (no problem for simple betterC stuff in wasm, just need to compile everything with the override). If LLVM can't cope with this format for a target's codegen, it'll error out. [I'm a fan of giving users full control and letting them explore the compiler/LLVM limits.]

---

[Plus there are some compiler assumptions wrt. `real` precision, in existing `TargetABI` implementations - as the `real` precision wasn't configurable so far.]

---

> [I'm a fan of giving users full control and letting them explore the compiler/LLVM limits.]

I see. Then that's fine by me. [Although LLVM is quite fragile when operating beyond its comfort zone]

---

`RealPrecision::DoubleDouble` should be handled here too (i.e. the user has set it explicitly)

---

> `RealPrecision::DoubleDouble` should be handled here too (i.e. the user has set it explicitly)

Right, and for all other architectures, it should throw an error.

---

> `RealPrecision::DoubleDouble` should be handled here too (i.e. the user has set it explicitly)

*That* I wanted to avoid though. I originally only wanted to expose the 2 standard IEEE formats as overrides, but unfortunately we need a way to make a native-PPC64 build with default IEEE quad ABI still cross-compile to the old doubledouble ABI. (I still don't wanna expose x87 if we can help it - although admittedly it could be useful for MSVC targets, to unlock the x87 `real`, which the compiler itself could use as `real_t` - bye-bye `dmd.root.longdouble`...)

But yeah, we can bail out as a compromise if this option is used for non-ppc64 targets.